internal/k8s/controllers/config_controller.go

```diff
@@ -149,6 +158,28 @@ func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 }
 
 func (r *ConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	p := predicate.NewPredicateFuncs(
+		func(obj client.Object) bool {
+			newNodeObj, ok := obj.(*corev1.Node)
+			if !ok {
+				return true
+			}
+			err := r.Get(context.Background(), types.NamespacedName{Namespace: newNodeObj.Namespace,
```

Comment:
I'd use the predicate version with the action instead of doing the get.

Reply:
done.